simulation/traffic_simulator/src/entity/vehicle_entity.cpp

```diff
@@ -221,7 +224,17 @@ void VehicleEntity::setAccelerationLimit(double acceleration)
     THROW_SEMANTIC_ERROR("Acceleration limit should be over zero.");
   }
   auto behavior_parameter = getBehaviorParameter();
-  behavior_parameter.acceleration = acceleration;
+  behavior_parameter.dynamic_constraints.max_acceleration = acceleration;
+  setBehaviorParameter(behavior_parameter);
+}
+
+void VehicleEntity::setAccelerationRateLimit(double acceleration_rate)
+{
+  if (acceleration_rate <= 0.0) {
+    THROW_SEMANTIC_ERROR("Acceleration limit should be over zero.");
+  }
+  auto behavior_parameter = getBehaviorParameter();
+  behavior_parameter.dynamic_constraints.max_acceleration_rate = acceleration_rate;
```

Comment:
Consider clamping the value to the range 0 to max_acceleration_rate instead of raising an error.
Also, the error message mentions acceleration instead of acceleration rate, which is inappropriate.

Reply:
Deceleration value should not under 0, because it is physically impossible.
So I think raising error is best.

Reply:
**the error message mentions acceleration instead of acceleration rate, which is inappropriate.**